packages/ai-jsx/src/core/log.ts

```diff
@@ -129,6 +137,14 @@ export class OpenTelemetryLogger extends LogImplementation {
       },
     });
   }
+
+  setAttribute(element: Element<any>, renderId: string, key: string, value: string): void {
+    // Set the attribute on the current span.
+    const span = opentelemetry.trace.getActiveSpan();
+    if (span) {
+      span.setAttribute(key, value);
+    }
+  }
```

Comment:
Heh, it's a bit funny that `element` isn't actually used in this method – do we actually want to commit ourselves to there only ever being one active span? And for that span only being linked to an element? Does that conflict with using a span to track the API call to the model provider? Maybe I'm missing something here.

Does this even work reliably with async?

Reply:
It does work with async -- OpenTelemetry uses https://nodejs.org/api/async_context.html to propagate context. (Which works nicely modulo https://github.com/open-telemetry/opentelemetry-js/issues/2951, hence the temporary workarounds for generators)

If we were worried about competing spans we could add our own property on the context for something like "active AI.JSX render span" to distinguish between "active span" but I think it'd be unusual for a component to go through the trouble of entering a custom span and use `logger.setAttribute` (with the specific expectation of it landing on the render span). So I'm going to defer that :)